classes/AddressFormat.php

```diff
@@ -236,11 +236,55 @@ public function checkFormatFields()
                     }
                 }
             }
+            $this->checkRequiredFields($usedKeyList);
         }
 
         return (count($this->_errorFormatList)) ? false : true;
     }
 
+    /**
+     * Checks that all required fields exist in a given fields list.
+     * Fills _errorFormatList array in case of absence of a required field.
+     *
+     * @param array $fieldList
+     */
+    protected function checkRequiredFields($fieldList)
+    {
+        foreach (self::getFieldsRequired() as $requiredField) {
+            if (!in_array($requiredField, $fieldList)) {
+                $this->_errorFormatList[] = $this->trans(
+                    'The field %s is required (in the %s tab).',
+                    array($requiredField, $this->getFieldTabName($requiredField)),
+                    'Admin.Notifications.Error');
```

Comment:
@LouiseBonnard  We are making this error message more specific, the name of the tab where the required field can be found will be displayed in it.

Reply:
Hi @matthieu-rolland, why are we adding the tab name?

Reply:
Because, if let's say lastname is missing, there are several "lastname" elements in the form, once in the Customer tab, another in the Address tab etc... We have several elements having the same name in different tabs like this. Otherwise it's not clear for the end user to know from where he can add the missing field.

![Capture d’écran 2019-09-30 à 16 47 20](https://user-images.githubusercontent.com/1784781/65889748-13542700-e3a2-11e9-9029-5882734b94c8.png)

Reply:
Alright then, thanks for the explanation. I would write: _The %s field (in tab %s) is required._

Reply:
@LouiseBonnard  thanks ! I made the modification